### PR TITLE
feat(config): hooks for applying middleware before/after graphql

### PR DESCRIPTION
**Pull request by deweyjose**

### CODE REVIEW

File: src/core/server.ts [hunk 1]
```diff
@@ -58,6 +59,7 @@ export class Server<C extends BaseContext> {
   logger: Logger;
   schema?: GraphQLSchema;
   bodyParserConfig?: OptionsJson;
+  expressApp: any;
```
Comment (pjpritch) [inline src/core/server.ts]: should this use the "express".Application as its type?

File: src/core/server.ts [hunk 2]
```diff
@@ -45,6 +44,8 @@ export interface ServerOptions<T> {
   warthogImportPath?: string;
   introspection?: boolean; // DEPRECATED
   bodyParserConfig?: OptionsJson;
+  onBeforeGraphqlHook?: (app: express.Application) => void;
```
Comment (goldcaddy77) [inline src/core/server.ts]: Should we call this `onBeforeGraphQLMiddleware`?

File: src/test/functional/server.test.ts [hunk 3]
```diff
@@ -329,6 +346,8 @@ describe('server', () => {
     }
     expect(error).toContain('Unable to find KitchenSink where');
   });
+
+  test('express hooks', async () => {});
```
Comment (goldcaddy77) [inline src/test/functional/server.test.ts]: Can we kill this?

**Comment by goldcaddy77** [comment]
This looks good - just one comment on naming.

One more thing, though - can you add this to the README?

File: src/core/server.ts (hunk 2 above)
Comment (pjpritch) [reply src/core/server.ts]: no.  this is a general purpose hook ...

File: src/test/functional/server.test.ts (hunk 3 above)
Comment (deweyjose) [reply src/test/functional/server.test.ts]: done

**Comment by deweyjose** [comment]
@goldcaddy77 updated readme

**Pull request merged**